cmd/juju/cloud/updatecredential.go

```diff
@@ -0,0 +1,134 @@
+
+Examples:
+    juju update-credential aws mysecrets
+
+See also: 
+    add-credential
+    credentials`[1:]
+
+type updateCredentialCommand struct {
+	modelcmd.ControllerCommandBase
+
+	api credentialAPI
+
+	cloud      string
+	credential string
+}
+
+// NewUpdateCredentialCommand returns a command to update credential details.
+func NewUpdateCredentialCommand() cmd.Command {
+	return modelcmd.WrapController(&updateCredentialCommand{})
+}
+
+// Init implements Command.Init.
+func (c *updateCredentialCommand) Init(args []string) error {
+	if len(args) < 2 {
+		return errors.New("Usage: juju update-credential <cloud-name> <credential-name>")
+	}
+	c.cloud = args[0]
+	c.credential = args[1]
+	return cmd.CheckEmpty(args[2:])
+}
+
+// Info implements Command.Info
+func (c *updateCredentialCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "update-credential",
+		Args:    "<cloud-name> <credential-name>",
+		Purpose: usageUpdateCredentialSummary,
+		Doc:     usageUpdateCredentialDetails,
+	}
+}
+
+// SetFlags implements Command.SetFlags.
+func (c *updateCredentialCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ControllerCommandBase.SetFlags(f)
+	f.StringVar(&c.credential, "credential", "", "Name of credential to update")
+	f.StringVar(&c.cloud, "cloud", "", "Cloud for which to update the credential")
+}
+
+type credentialAPI interface {
+	UpdateCredential(tag names.CloudCredentialTag, credential jujucloud.Credential) error
+	Clouds() (map[names.CloudTag]jujucloud.Cloud, error)
+	Close() error
+}
+
+func (c *updateCredentialCommand) getAPI() (credentialAPI, error) {
+	if c.api != nil {
+		return c.api, nil
+	}
+	api, err := c.NewAPIRoot()
+	if err != nil {
+		return nil, errors.Annotate(err, "opening API connection")
+	}
+	return apicloud.NewClient(api), nil
+}
+
+// Run implements Command.Run
+func (c *updateCredentialCommand) Run(ctx *cmd.Context) error {
+	cred, err := c.ClientStore().CredentialForCloud(c.cloud)
+	if errors.IsNotFound(err) {
+		ctx.Infof("No credentials exist for cloud %q", c.cloud)
+		return nil
+	} else if err != nil {
+		return err
+	}
+	credToUpdate, ok := cred.AuthCredentials[c.credential]
+	if !ok {
+		ctx.Infof("No credential called %q exists for cloud %q", c.credential, c.cloud)
+		return nil
+	}
+
+	accountDetails, err := c.ClientStore().AccountDetails(c.ControllerName())
+	if err != nil {
+		return errors.Trace(err)
+	}
+	credentialTag, err := common.ResolveCloudCredentialTag(
+		names.NewUserTag(accountDetails.User), names.NewCloudTag(c.cloud), c.credential,
+	)
+
+	client, err := c.getAPI()
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	if err := client.UpdateCredential(credentialTag, credToUpdate); err != nil {
+		return c.processError(client, err)
+	}
+	ctx.Infof("Updated credential %q for user %q on cloud %q.", c.credential, accountDetails.User, c.cloud)
+	return nil
+}
+
+func (c *updateCredentialCommand) processError(client credentialAPI, initialErr error) error {
```

Comment:
-1, this assumes to much. Please either use a special error code or do the check in the client first.